edit/harvesterhci.io.virtualmachineimage.vue

```diff
@@ -80,6 +87,30 @@ export default {
 
       this.value.spec.url = url;
       if (VM_IMAGE_FILE_FORMAT.includes(fileSuffiic)) {
+        if (fileSuffiic) {
+          Object.keys(this.value.labels).forEach((key, idx) => {
+            if (key === HCI_ANNOTATIONS.IMAGE_SUFFIC) {
+              this.$refs.labels.remove(idx);
+            }
+          });
+          if (fileSuffiic === 'iso') {
+            this.$refs.labels.add(HCI_ANNOTATIONS.IMAGE_SUFFIC, fileSuffiic);
+          } else {
+            this.$refs.labels.add(HCI_ANNOTATIONS.IMAGE_SUFFIC, rawORqcow2);
+          }
+
+          const os = this.getOSType(url);
+
+          if (os) {
+            Object.keys(this.value.labels).forEach((key, idx) => {
+              if (key === HCI_ANNOTATIONS.OS_TYPE) {
+                this.$refs.labels.remove(idx);
+              }
+            });
+            this.$refs.labels.add(HCI_ANNOTATIONS.OS_TYPE, os.value);
+          }
```

Comment:
Is the os has reasons to put it under the `if (VM_IMAGE_FILE_FORMAT.includes(fileSuffiic))` condition?

Reply:
Done.